geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/HAEventWrapper.java

```diff
@@ -411,4 +395,15 @@ public long decAndGetReferenceCount() {
     return rcUpdater.decrementAndGet(this);
   }
 
+  public long incrementPutInProgressCounter() {
+    return putInProgressCountUpdater.incrementAndGet(this);
+  }
+
+  public long decrementPutInProgressCounter() {
```

Comment:
Consider logging if < 0

Reply:
We don't currently do this for the existing ref count and we log the ref count when debug is enabled.  Therefore, we decided this isn't necessary.